### ISSUE TITLE
[SAM3.1] Interactive loader silently skips critical multiplex checkpoint weights

### ISSUE THREAD

**Issue by HowcanoeWang** [ultralytics/ultralytics#25829]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Predict

### Bug

`SAM3Predictor` does not correctly load `sam3.1_multiplex.pt` for interactive image prompting.

In `build_sam3.py::_load_checkpoint(..., interactive=True)`, the multiplex keys are not mapped to the constructed interactive model names. The final `load_state_dict(..., strict=False)` silently leaves critical modules randomly initialized:

- interactive neck: `0/22` target tensors loaded
- prompt encoder: `0/17`
- mask decoder: `0/131`
- `no_mem_embed`: not loaded

Only `406/741` target tensors are loaded in total. Box-prompt inference then produces a dense periodic/checkerboard binary mask instead of the requested object.

With a local alias mapping and shape validation, all `573` tensors consumed by the current single-image path load successfully and the same predictor, image, and box prompt produce a coherent grape mask.

<img width="2685" height="642" alt="Image" src="https://github.com/user-attachments/assets/0d89d54d-eaa8-46fe-b241-4b76167872b4" />

Expected behavior: required interactive weights should either load correctly or produce a clear loading error, rather than continuing with random weights.

### Environment

```text
Ultralytics 8.4.120
Python 3.12.12
PyTorch 2.12.1+cu130
CUDA 13.0
NVIDIA GeForce RTX 3090, 24 GiB
Linux x86-64
```

The same checkpoint-schema problem was first observed with Ultralytics 8.4.89.

### Minimal Reproducible Example

The attached ZIP contains a locked `uv` environment, checkpoint verification, test image, and an executed notebook showing both the official failure and corrected result. It does not redistribute the gated model weight.

First request access to the official Meta repository, accept its terms, and download `sam3.1_multiplex.pt`:

<https://huggingface.co/facebook/sam3.1/tree/main>

Extract the attached ZIP, place `sam3.1_multiplex.pt` in its root beside the notebook, then run from that directory:

```bash
unzip ultralytics_sam31_multiplex_loader_repro.zip
cd ultralytics_sam31_multiplex_loader_repro
cp /path/to/downloaded/sam3.1_multiplex.pt .
uv sync --frozen
uv run jupyter execute ultralytics_sam31_multiplex_loader_repro.ipynb \
  --inplace --timeout=900
```

The notebook performs no model download and stops with a clear error if the local checkpoint is missing or does not match the expected identity.

Checkpoint identity:

```text
Size: 3,502,755,717 bytes
SHA-256: 0567debeec80ba4ac6369540c6c248025283cb3ff2b92827509e57e2b3541cb6
```

[2026081612_ultralytics_sam31_multiplex_loader_repro.zip](https://github.com/user-attachments/files/31113265/2026081612_ultralytics_sam31_multiplex_loader_repro.zip)

### Additional

The likely fix is to map `interactive_convs -> sam2_convs`, `interactive_sam_prompt_encoder -> sam_prompt_encoder`, `interactive_sam_mask_decoder -> sam_mask_decoder`, and `interactivity_no_mem_embed -> no_mem_embed`, then validate required keys and tensor shapes before inference.

The attached notebook captures the official loader's actual missing/unexpected keys and keeps all Ultralytics construction, loading, inference, and result-extraction code visible inline.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @HowcanoeWang, thank you for the detailed report and reproducible notebook. The evidence that critical SAM3.1 interactive checkpoint weights are silently skipped during `SAM3Predictor` loading is very helpful. An Ultralytics engineer will review the checkpoint key mapping and required-key/shape validation soon. This is an automated response.

**Comment by Y-T-G** [comment]
Duplicate of https://github.com/ultralytics/ultralytics/issues/25152

**Comment by Y-T-G** [comment]
We do not support SAM3.1 yet in Ultralytics
